fishtank/src/cluster.ts

```diff
@@ -51,20 +57,53 @@ export class Cluster {
       labels: { [CLUSTER_LABEL]: this.name },
     }
 
+    const args: string[] = []
+
     if (options.config) {
       const configString = JSON.stringify(options.config)
 
-      const dest = join(tmpdir(), 'fishtank', containerName, '.ironfish')
+      const dest = join(tmpdir(), 'fishtank', this.name, options.name, '.ironfish')
       await promises.mkdir(dest, {
         recursive: true,
       })
 
       await promises.writeFile(resolve(dest, 'config.json'), configString)
 
-      runOptions.volumes = new Map<string, string>([[dest, '/root/.ironfish']])
+      if (runOptions.volumes === undefined) {
+        runOptions.volumes = new Map<string, string>([[dest, DEFAULT_CONTAINER_DATADIR]])
+      } else {
+        runOptions.volumes.set(dest, DEFAULT_CONTAINER_DATADIR)
+      }
+    }
+
+    if (options.networkDefinition) {
+      const networkDefinitionString = JSON.stringify(options.networkDefinition)
+
+      const dest = join(tmpdir(), 'fishtank', this.name, options.name, '.ironfish')
+      await promises.mkdir(dest, {
+        recursive: true,
+      })
+
+      await promises.writeFile(resolve(dest, 'customNetwork.json'), networkDefinitionString)
+
+      let containerPath = DEFAULT_CONTAINER_DATADIR
+      if (runOptions.volumes === undefined) {
+        runOptions.volumes = new Map<string, string>([[dest, containerPath]])
+      } else if (runOptions.volumes.has(dest)) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        containerPath = runOptions.volumes.get(dest)!
+      } else {
+        runOptions.volumes.set(dest, containerPath)
+      }
+      args.push(`--customNetwork=${resolve(containerPath, 'customNetwork.json')}`)
```

Comment:
I *think* that if you call this `network.json`, then no new arguments are required

Reply:
No, see here https://github.com/iron-fish/ironfish/blob/022ebac2cb58d2c384cff6222f1d7baaf3e5100f/ironfish/src/networkDefinition.ts#L85